lib/parser.js

```diff
@@ -1189,7 +1231,11 @@ function parseStatement(parseContext) {
   // if there's more tokens, split them by a ' ' similar to how joinTokens would work
   if(tokens.length > 0)
   {
-    statement += ' ';
+    // peek at the next token and don't apply space since it would break
+    if(shouldSeparateTokens(tokens))
```

Comment:
This doesn't seem to be necessary, but it does make the output slightly nicer

Reply:
ah, does `document . window` work? i'll double check on the console

Reply:
```
$ node
> x = {
... a: 5
... }
{ a: 5 }
> x . a
5
>

```

I could go either way, without the separate token bit it just makes things slightly simpler

Reply:
@vpzomtrrfrt any preference? looks like the beautyfier would actually combine those, so my vote is no-fancy thing and just let it separate them!

Reply:
Since it simplifies our stuff, i'm gonna get rid of the check !